build.rs

```diff
@@ -23,13 +27,32 @@ fn main() {
   }
 
   // Builds hvm.cu
-  if std::process::Command::new("nvcc").arg("--version").stdout(std::process::Stdio::null()).stderr(std::process::Stdio::null()).status().is_ok() {
+  if Command::new("nvcc").arg("--version").stdout(std::process::Stdio::null()).stderr(std::process::Stdio::null()).status().is_ok() {
     if let Ok(cuda_path) = std::env::var("CUDA_HOME") {
       println!("cargo:rustc-link-search=native={}/lib64", cuda_path);
     } else {
       println!("cargo:rustc-link-search=native=/usr/local/cuda/lib64");
     }
 
+    // Compile get_shared_mem.cu
+    if let Ok(output) = Command::new("nvcc")
+      .args(&["src/get_shared_mem.cu", "-o", "get_shared_mem"])
+      .output()
+      .and_then(|_| Command::new("./get_shared_mem").output()) {
+        if output.status.success() {
+          let shared_mem_str = String::from_utf8_lossy(&output.stdout).trim().to_string();
+          std::fs::write("src/shared_mem_config.h", format!("#define HVM_SHARED_MEM {}", shared_mem_str))
```

Comment:
Instead of writing it to a build generated file, can't we just set a define flag, like we do for the number of cores? It's prpbably possible with nvcc as well

Reply:
true, that sounds cleaner indeed, will change it to that